src/modules/userMethods.ts

```diff
@@ -460,4 +461,124 @@ export const userMethods = {
+                    toJSON: () => [
+                        {
+                            serviceName: 'Phone',
+                            serviceId: data.phone,
+                        },
+                        {
+                            serviceName: 'Accounting system',
+                            serviceId: data.accountingId,
+                        },
+                    ],
+                },
+            },
+        });
+    },
+
+    getUsersRequests: async <
+        T extends UserCreationRequest & {
+            status: 'Approved' | 'Denied' | null;
+            supervisor: User;
+            organization: OrganizationUnit;
+            group: Group;
+            services: Record<'serviceName' | 'serviceId', string>[] | null;
+        },
+    >(): Promise<T[]> => {
+        const requests = await prisma.userCreationRequest.findMany({
+            where: {
+                status: null,
+            },
+            include: {
+                group: true,
+                organization: true,
+                supervisor: true,
+            },
+        });
+
+        return requests as T[];
+    },
+
+    declineUserRequest: async ({ id, comment }: { id: string; comment?: string }) => {
+        return prisma.userCreationRequest.update({
+            where: { id },
+            data: { status: 'Denied', comment },
+        });
+    },
+
+    acceptUserRequest: async ({ id, comment }: { id: string; comment?: string }) => {
+        const acceptedRequest = await prisma.userCreationRequest.update({
+            where: { id },
+            data: { status: 'Approved', comment },
+            include: { supervisor: true },
+        });
+
+        const services = acceptedRequest.services as { serviceId: string; serviceName: string }[];
+
+        const checkUser = await userMethods.getByLogin(acceptedRequest.login);
+
+        if (checkUser) {
+            throw new Error('User already exists');
+        }
+
+        const newUser = await prisma.user.create({
+            data: {
+                name: acceptedRequest.name,
+                email: acceptedRequest.email,
+                supervisorId: acceptedRequest.supervisor.id,
+                login: acceptedRequest.login,
+                memberships: { create: { groupId: acceptedRequest.groupId } },
+                organizationUnitId: acceptedRequest.organizationUnitId,
+                services: { createMany: { data: services } },
+            },
+        });
+
+        if (newUser && acceptedRequest.createExternalAccount) {
+            const [surname, firstName, middleName] = acceptedRequest.name.split(' ');
+
+            const phone = services.find((service) => service.serviceName === 'Phone')?.serviceId;
+            const accountingId = services.find((service) => service.serviceName === 'Accounting system')?.serviceId;
+
+            if (!phone) {
+                throw new Error('Phone service not found');
+            }
+
+            await externalUserCreate({
```

Comment:
External user is created before crew user, if external fails we shouldn't create crew user